apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1207,7 +1216,44 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = Queries.add_old_interfaces(db_client, new_state.device_id, old_introspection)
     :ok = Queries.remove_old_interfaces(db_client, new_state.device_id, readded_introspection)
 
-    # TODO: handle triggers for interface minor updates
+    interface_minor_updated_map =
+      Enum.map(old_minors, fn {iface, old_minor} ->
+        old_major = Map.fetch!(state.introspection, iface)
+        {iface, old_major, old_minor}
+      end)
+      |> Enum.filter(fn {iface, old_major, _old_minor} ->
+        Map.get(db_introspection_map, iface) == old_major
+      end)
+      |> Enum.reduce(%{}, fn {iface, old_major, old_minor}, acc ->
+        new_minor = Map.get(db_introspection_minor_map, iface)
+
+        if new_minor != old_minor do
+          Map.put(acc, {iface, old_major}, {old_minor, new_minor})
+        else
+          acc
+        end
+      end)
+
+    Enum.each(interface_minor_updated_map, fn {{interface_name, interface_major},
+                                               {old_minor, new_minor}} ->
+      interface_minor_update_targets =
+        Map.get(
+          device_triggers,
+          {:on_interface_minor_updated, CQLUtils.interface_id(interface_name, interface_major)},
+          []
+        )
+
+      TriggersHandler.interface_minor_updated(
+        interface_minor_update_targets,
+        realm,
+        device_id_string,
+        interface_name,
+        interface_major,
+        old_minor,
+        new_minor,
+        timestamp_ms
+      )
+    end)
```

Comment:
You're building the map just to iterate on it with `Enum.each` and then throwing it away, you can replace this whole chunk _more elegantly_ with a `for` comprehension:
```elixir
    for {interface_name, old_minor} <- old_minors,
        interface_major = Map.fetch!(state.introspection, interface_name),
        Map.get(db_introspection_map, interface_name) == interface_major,
        new_minor = Map.get(db_introspection_minor_map, interface_name),
        new_minor != old_minor do
      interface_id = CQLUtils.interface_id(interface_name, interface_major)

      Map.get(device_triggers, {:on_interface_minor_updated, interface_id}, [])
      |> TriggersHandler.interface_minor_updated(
        realm,
        device_id_string,
        interface_name,
        interface_major,
        old_minor,
        new_minor,
        timestamp_ms
      )
    end
```